django/contrib/admin/views/main.py

```diff
@@ -57,16 +79,20 @@ def __init__(self, request, model, list_display, list_display_links,
         self.sortable_by = sortable_by
 
         # Get search parameters from the query string.
-        try:
-            self.page_num = int(request.GET.get(PAGE_VAR, 0))
-        except ValueError:
-            self.page_num = 0
+        change_list_form = ChangeListForm(request.GET)
+        if not change_list_form.is_valid():
+            raise forms.ValidationError(change_list_form.errors)
```

Comment:
I think this is problematic as it will result in an unhandled exception instead of displaying an error to the user.

Reply:
I try to find a sample code about it but I didn't. Can we add an error message into django.contrib.messages with a specific key? we would parse this error message and cover searchbar(search_form.html) input with error field tag.

Reply:
Using messages might work.